src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -129,7 +129,7 @@ private CompilationWithAnalyzers(Compilation compilation, ImmutableArray<Diagnos
             _cancellationToken = cancellationToken;
 
             _compilationData = new CompilationData(_compilation);
-            _analysisState = new AnalysisState(analyzers, _compilationData);
+            _analysisState = new AnalysisState(analyzers, _compilationData, _compilation.Options);
```

Comment:
compilation data itself seems not hold onto compilation but it holds onto suppression state which internally hold onto compilation. so, looks like you don't need to add one more parameter.